On Mon, Nov 20, 2000 at 12:51:34PM +0100, Arkadiusz Miskiewicz wrote:
> Hi,
>
> lftp doesn't follow rfc2732:
>
> [root@arm /root]# LANG=C lftp ftp://[3ffe:8010:34:17::1:1]/pub
> lftp: [3ffe:8010:34:17::1:1]: Unknown host

Is this patch good enough?

Index: url.cc
===================================================================
RCS file: /home/lav/cvsroot/lftp/src/url.cc,v
retrieving revision 1.16
diff -u -r1.16 url.cc
--- url.cc	2000/09/29 07:51:22	1.16
+++ url.cc	2000/11/20 13:11:57
@@ -144,8 +144,24 @@
    // extract host name/password
    scan=base;
    host=base;
-   while(*scan && *scan!=':')
-      scan++;
+   if(*scan=='[') // RFC2732 [ipv6]
+   {
+      while(*scan && *scan!=']')
+	 scan++;
+      if(*scan==']')
+      {
+	 *scan++=0;
+	 host++;
+      }
+      else
+	 scan=base;
+   }
+
+   if(scan==base)
+   {
+      while(*scan && *scan!=':')
+	 scan++;
+   }

    if(*scan==':') // port found
    {

--
   Alexander.                      | never mind that noise you heard  
